server/src/main/java/org/elasticsearch/search/query/QueryPhase.java

```diff
@@ -204,6 +224,32 @@ static boolean execute(SearchContext searchContext,
                 hasFilterCollector = true;
             }
 
+            CheckedConsumer<List<LeafReaderContext>, IOException> leafSorter = l -> {};
+            // try to rewrite numeric or date sort to the optimized distanceFeatureQuery
+            if ((searchContext.sort() != null) && SYS_PROP_LONG_SORT_OPTIMIZED) {
+                Query rewrittenQuery = tryRewriteLongSort(searchContext, searcher.getIndexReader(), query, hasFilterCollector);
+                if (rewrittenQuery != null) {
+                    query = rewrittenQuery;
+                    // modify sorts: add sort on _score as 1st sort, and move the sort on the original field as the 2nd sort
+                    SortField[] oldSortFields = searchContext.sort().sort.getSort();
+                    DocValueFormat[] oldFormats = searchContext.sort().formats;
+                    SortField[] newSortFields = new SortField[oldSortFields.length + 2];
+                    DocValueFormat[] newFormats = new DocValueFormat[oldSortFields.length + 2];
+                    newSortFields[0] = SortField.FIELD_SCORE;
+                    newFormats[0] = DocValueFormat.RAW;
+                    // Add a tiebreak on _doc in order to be able to search
+                    // the leaves in any order. This is needed since we reorder
+                    // the leaves based on the minimum/maxim value in each segment.
+                    newSortFields[newSortFields.length-1] = SortField.FIELD_DOC;
```

Comment:
This is not needed anymore since we use the shared collector manager ?

Reply:
Thanks Jim, indeed shared collector manager with tie on scores [compares by docID ](https://github.com/apache/lucene-solr/blob/master/lucene/core/src/java/org/apache/lucene/search/MaxScoreAccumulator.java#L68). I will correct this.

Reply:
@jimczi  addressed in the last commit